Yosemite/YosemiteTests/Stores/BlazeStoreTests.swift

```diff
@@ -379,6 +383,141 @@ final class BlazeStoreTests: XCTestCase {
         XCTAssertTrue(result.isFailure)
         XCTAssertEqual(result.failure as? NetworkError, .timeout())
     }
+
+    // MARK: - Synchronize target topics
+
+    func test_synchronizeTargetTopics_is_successful_when_fetching_successfully() throws {
+        // Given
+        remote.whenFetchingTargetTopics(thenReturn: .success([.fake().copy(id: "IAB1")]))
+        let store = BlazeStore(dispatcher: Dispatcher(),
+                               storageManager: storageManager,
+                               network: network,
+                               remote: remote)
+
+        // When
+        let result = waitFor { promise in
+            store.onAction(BlazeAction.synchronizeTargetTopics(siteID: self.sampleSiteID, locale: "en", onCompletion: { result in
+                promise(result)
+            }))
+        }
+
+        //Then
+        let topics = try result.get()
+        XCTAssertEqual(topics.count, 1)
+    }
+
+    func test_synchronizeTargetTopics_stores_topics_upon_success() throws {
+        // Given
+        remote.whenFetchingTargetTopics(thenReturn: .success([.fake().copy(id: "IAB1")]))
+        let store = BlazeStore(dispatcher: Dispatcher(),
+                               storageManager: storageManager,
+                               network: network,
+                               remote: remote)
+        XCTAssertEqual(storedTargetTopicCount, 0)
+
+        // When
+        let result = waitFor { promise in
+            store.onAction(BlazeAction.synchronizeTargetTopics(siteID: self.sampleSiteID, locale: "en", onCompletion: { result in
+                promise(result)
+            }))
+        }
+
+        // Then
+        XCTAssertTrue(result.isSuccess)
+        XCTAssertEqual(storedTargetTopicCount, 1)
+    }
+
+    func test_synchronizeTargetTopics_overwrites_existing_devices_with_the_given_locale() throws {
```

Comment:
```suggestion
    func test_synchronizeTargetTopics_overwrites_existing_topics_with_the_given_locale() throws {
```

Reply:
Fixed in a1c5661cbccb30f7facd106f22a005ff2487236c.